flake.nix

```diff
@@ -9,15 +9,18 @@
     lib = {
       hm = import ./modules/lib { lib = nixpkgs.lib; };
       homeManagerConfiguration = { configuration, system, homeDirectory
-        , username, extraSpecialArgs ? { }
+        , username, extraModules ? [ ], extraSpecialArgs ? { }
         , pkgs ? builtins.getAttr system nixpkgs.outputs.legacyPackages
         , check ? true }@args:
         import ./modules {
-          inherit pkgs check extraSpecialArgs;
+          inherit pkgs check;
+
           configuration = { ... }: {
-            imports = [ configuration ];
+            imports = [ configuration ] ++ extraModules;
             home = { inherit homeDirectory username; };
           };
+
+          extraSpecialArgs = extraSpecialArgs // { inherit pkgs; };
```

Comment:
I'm honestly surprised this is required.

Reply:
Yeah, me too, but really tied to juggle with the option `useNixpkgsModule` and `config.home.stateVersion = "20.09"`. But nothing worked. But as soon as I added it to `extraSpecialArgs` it worked without a problem.